pilot/pkg/model/sidecar.go

```diff
@@ -89,6 +89,7 @@ type SidecarScope struct {
 	// Union of services imported across all egress listeners for use by CDS code.
 	services           []*Service
 	servicesByHostname map[host.Name]*Service
+	servicesByNames    map[ConfigKey]*Service
```

Comment:
not clear why we need this when we already have `servicesByHostname`? If we do have a need we should very clearly explain in the code comments why we need both